cores/esp8266/WString.h

```diff
@@ -263,8 +263,8 @@ class String {
         void setCapacity(int cap) { if (!isSSO()) ptr.cap = cap; }
         void setBuffer(char *buff) { if (!isSSO()) ptr.buff = buff; }
         // Buffer accessor functions
-        const char *buffer() const { return wbuffer(); }
-        char *wbuffer() const { return isSSO() ? const_cast<char *>(sso.buff) : ptr.buff; } // Writable version of buffer
+        const char *buffer() const { return isSSO() ? sso.buff : ptr.buff; }
+        char *wbuffer() { return const_cast<char *>(buffer()); } // Writable version of buffer
```

Comment:
just c/p? I though the idea was to also remove the const_cast, not just fix the const-correctness of the class itself

core guidelines also suggest the same approach https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es50-dont-cast-away-const
(and since we can't have a specific member that will hold the pointer)

Reply:
Oh ok, but the const correctness is still there, since wbuffer is a non const function it ensures the internal buffer actually is writable, so it just avoids copy paste without any risk. But yes I can copy and paste.

Reply:
@mcspr should I copy and paste or are you ok with my reasoning?

Reply:
> The problem wasn't the const_cast, it was that wbuffer was a const function so other const functions were using it to write to String, which could be in const storage. Now if wbuffer is called, String is ensured to not be in const storage.

I concur.

Re-reading the paragraph on the order, it's the correct thing to do.
But don't really follow the reason behind the explicit `static_cast<const Foo&>(*this).get_bar()` (ref. core guidelines) to call the method instead of just get_bar()